plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -137,6 +137,7 @@ def get_token(module_params):
         client_secret = module_params.get('auth_client_secret')
         connection_timeout = module_params.get('connection_timeout')
         auth_url = URL_TOKEN.format(url=base_url, realm=auth_realm)
+        headers = {"User-Agent": "Ansible"}
```

Comment:
Instead of explicitly setting this header, use the `http_agent` option of `open_url()`.

Also, it might make sense to make this configurable instead of hard-coding a value.

Reply:
Sure, I'll try that too, but what I suggest is to implicitly configure ansible urls module to send a User-Agent header With value `Ansible <version>` so that all the requests sent over web have it, not just the Keycloak library, 
Curl already does that, and that's the reason with the mentioned scenario for Keycloak, curl request from cli went successful, because it sends an implicit User-Agent header.

We definitely can make it configurable via module params, but a sane default would ensure that user agent is identified. It has no significance over the requests, but is mostly hepful for SEO Analytics and Logging, to identify the type of application client its originating from, and its would be preferable that all the requests sent with any ansible module using urllib or more abstract urls module to have it in order to identify.